release

```diff
@@ -0,0 +1,154 @@
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+MAKEFILE_START_LINE=$(\
+  grep -ne '^# 8><' "$0" \
+  | tail -n 1 \
+  | awk -F: '{print $1}')
+
+LITANI_RELEASE=/tmp/litani-release
+mkdir -p "${LITANI_RELEASE}"
+
+MAKEFILE=$(mktemp ${LITANI_RELEASE}/XXXXXX || exit 1)
+tail -n+${MAKEFILE_START_LINE} "$0" > "${MAKEFILE}"
+
+make -Bj 1 -f "${MAKEFILE}"
+SUCCESS="$?"
+
+exit "${SUCCESS}"
+
+# 8>< - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+#
+# Makefile starts here
+
+TMP=/tmp/litani-release
+
+create-release: \
+	$(TMP)/current-release-number \
+	$(TMP)/rc-number \
+	$(TMP)/commits.json \
+	merge-develop-release \
+	commit-changelog \
+	tag-release \
+	create-rc
+
+create-rc: $(TMP)/rc-number
+	git checkout -B develop
+	new_minor=$$(cat $(TMP)/rc-number); \
+	  sed \
+	    -e "s/^VERSION_MINOR.+/VERSION_MINOR = $${new_minor}/g" \
+	    -I '' lib/litani.py
+	sed -e 's/^RC *= *False/RC = True/g' -I '' lib/litani.py
+	git add lib/litani.py
+	git commit -m "Create version 1.$$(cat $(TMP)/rc-number) release candidate"
+
+$(TMP)/rc-number:
+	printf "%d\n" \
+	  $$(( $$(grep -e '^VERSION_MINOR' lib/litani.py | awk '{print $$3}') + 1 )) \
+	  > $@
+
+$(TMP)/current-release-number:
+	printf "%d.%d.%d\n" \
+	  $$(grep -e '^VERSION_MAJOR' lib/litani.py | awk '{print $$3}') \
+	  $$(grep -e '^VERSION_MINOR' lib/litani.py | awk '{print $$3}') \
+	  $$(grep -e '^VERSION_PATCH' lib/litani.py | awk '{print $$3}') \
+	  > "$@"
+
+merge-develop-release: $(TMP)/current-release-number
+	git checkout -q release
+	git merge \
+	  --no-ff \
+	  --message="Bump version to $$(cat $(TMP)/current-release-number)" \
+	  develop
+
+commit-changelog: $(TMP)/updated-changelog
+	sed -e 's/^RC *= *True/RC = False/g' -I '' lib/litani.py
+	cp $(TMP)/updated-changelog CHANGELOG
+	git add lib/litani.py CHANGELOG
+	git commit --amend --reuse-message=HEAD
+
+$(TMP)/updated-changelog: \
+	    $(TMP)/current-release-number \
+	    $(TMP)/user-changelog \
+	    $(TMP)/old-changelog
+	printf "                              CHANGELOG\n" > $@
+	printf '                              `````````\n\n' >> $@
+	printf "Version %s -- %s\n" \
+	  $$(cat $(TMP)/current-release-number) \
+	  $$(date +%Y-%m-%d) >> $@
+	printf -- "----------------------------\n" >> $@
+	grep -ve '^#' $(TMP)/user-changelog >> $@
+	cat $(TMP)/old-changelog >> $@
+
+$(TMP)/old-changelog:
+	start_line=$$(\
+	  grep -ne '^Version ' CHANGELOG \
+	  | head -n 1 \
+	  | awk -F: '{print $$1}' \
+	); \
+	tail -n+$${start_line} CHANGELOG > $@
+
+$(TMP)/user-changelog: $(TMP)/commits.json
+	printf "" > $@
+	printf "%s\n" "# This will be the changelog entry for this release." >> $@
+	printf "%s\n" "# Lines beginning with '#' will be ignored." >> $@
```

Comment:
When I ran this locally, I left only one newline in the end of the user-changelog whereas I should have left two newlines in order for the updated-changelog to maintain the standard "two newlines" separating sections about different versions.

Should we add `	printf "%s\n" "# Leave 2 newlines following the last entry." >> $@` ?

Reply:
I fixed the code so that it discards whatever blanks the user added at the beginning and end of the file, after stripping out comments, and then adds two more blanks.